File: Samples/CslaLight/vb/RemotePortalWithDesignTimeBlendSupport/Sample.Business.Server/BusinessClasses/ReadOnlyCompany.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Csla
Imports Csla.Security
Imports Csla.Core
Imports Csla.Serialization
Imports Csla.Silverlight
Imports Csla.Validation
Imports System.ComponentModel

#If (Not SILVERLIGHT) Then
Imports Csla.Data
Imports System.Data.SqlClient
#End If

Namespace Sample.Business
  <Serializable()> _
  Public Class ReadOnlyCompany
    Inherits ReadOnlyBase(Of ReadOnlyCompany)

#If SILVERLIGHT Then
    Public Sub New()
    End Sub
#Else
    Private Sub New()
    End Sub
#End If


    Private Shared CompanyIdProperty As PropertyInfo(Of Integer) = RegisterProperty(Of Integer)(New PropertyInfo(Of Integer)("CompanyId", "Company Id", 0))
    Public ReadOnly Property CompanyId() As Integer
      Get
        Return GetProperty(CompanyIdProperty)
      End Get
    End Property

    Private Shared CompanyNameProperty As PropertyInfo(Of String) = RegisterProperty(Of String)(New PropertyInfo(Of String)("CompanyName", "Company Name", String.Empty))
    Public ReadOnly Property CompanyName() As String
      Get
        Return GetProperty(CompanyNameProperty)
      End Get
    End Property

    Protected Overrides Sub AddAuthorizationRules()
      Dim canRead() As String = {"AdminUser", "RegularUser", "ReadOnlyUser"}
      AuthorizationRules.AllowGet(GetType(ReadOnlyCompany), canRead)
      AuthorizationRules.AllowRead(CompanyNameProperty, canRead)
      AuthorizationRules.AllowRead(CompanyIdProperty, canRead)
    End Sub

    <ComponentModel.EditorBrowsable(ComponentModel.EditorBrowsableState.Never)> _
    Public Shared Function DesignTime_Create(ByVal name As String) As ReadOnlyCompany
      Dim returnValue As New ReadOnlyCompany
      returnValue.LoadProperty(CompanyNameProperty, name)
      Return returnValue
    End Function

#If (Not Silverlight) Then

    Public Shared Function GetReadOnlyCompany(ByVal reader As SafeDataReader) As ReadOnlyCompany
      Return DataPortal.FetchChild(Of ReadOnlyCompany)(reader)
    End Function

    Private Sub Child_Fetch(ByVal reader As SafeDataReader)
      LoadProperty(Of Integer)(CompanyIdProperty, reader.GetInt32("CompanyId"))
      LoadProperty(Of String)(CompanyNameProperty, reader.GetString("CompanyName"))
    End Sub


#End If
  End Class
End Namespace

File: Samples/CslaLight/cs/RemotePortalWithDesignTimeBlendSupport/Sample.Business.Server/BusinessClasses/ReadOnlyCompany.cs
//INSTANT C# NOTE: Formerly VB.NET project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;

using System.Text;
using Csla;
using Csla.Security;
using Csla.Core;
using Csla.Serialization;
using Csla.Silverlight;
using Csla.Validation;
using System.ComponentModel;

#if !SILVERLIGHT
using Csla.Data;
using System.Data.SqlClient;
#endif

namespace Sample.Business
{
  [Serializable()]
  public class ReadOnlyCompany : ReadOnlyBase<ReadOnlyCompany>
  {

#if SILVERLIGHT
    public ReadOnlyCompany()
    {
    }
#else
	private ReadOnlyCompany()
	{
	}
#endif


    private static PropertyInfo<int> CompanyIdProperty = RegisterProperty<int>(new PropertyInfo<int>("CompanyId", "Company Id", 0));
    public int CompanyId
    {
      get
      {
        return GetProperty(CompanyIdProperty);
      }
    }

    private static PropertyInfo<string> CompanyNameProperty = RegisterProperty<string>(new PropertyInfo<string>("CompanyName", "Company Name", string.Empty));
    public string CompanyName
    {
      get
      {
        return GetProperty(CompanyNameProperty);
      }
    }

    protected override void AddAuthorizationRules()
    {
      string[] canRead = { "AdminUser", "RegularUser", "ReadOnlyUser" };
      AuthorizationRules.AllowGet(typeof(ReadOnlyCompany), canRead);
      AuthorizationRules.AllowRead(CompanyNameProperty, canRead);
      AuthorizationRules.AllowRead(CompanyIdProperty, canRead);
    }

    [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]
    internal static ReadOnlyCompany DesignTime_Create(string name)
    {
      ReadOnlyCompany returnValue = new ReadOnlyCompany();
      returnValue.LoadProperty(CompanyNameProperty, name);
      return returnValue;
    }

#if ! SILVERLIGHT

	public static ReadOnlyCompany GetReadOnlyCompany(SafeDataReader reader)
	{
	  return DataPortal.FetchChild<ReadOnlyCompany>(reader);
	}

	private void Child_Fetch(SafeDataReader reader)
	{
	  LoadProperty<int>(CompanyIdProperty, reader.GetInt32("CompanyId"));
	  LoadProperty<string>(CompanyNameProperty, reader.GetString("CompanyName"));
	}


#endif
  }
}
